msg.wrap {text}

msg.config.unable_to_read_file Failed to load configuration (using defaults): "{error}"

msg.debug.saving_to_path Saving a debug log to {path}, please check it for sensitive information before sharing it.
msg.debug.unable_to_write_file Unable to write to our debug log file ({path}): {error}

msg.info.running_on [INFO] Erebus is running on port {port} with {mode} mode.
msg.info.server [INFO] Erebus server is located at {protocol}://{address}:{port}

msg.notice.control_conn_closed Tor control connection was closed.
msg.notice.try_to_connect Unable to establish a connection with tor control. Retrying in {interval} seconds.
msg.notice.new_connection New connection with Tor established.

msg.warn.unable_to_attach Unable to attach listeners: {reason}

msg.ws.add_error Undefined type ({type}) when adding websocket.
msg.ws.remove_error Undefined type ({type}) when removing websocket.
msg.ws.get_error Undefined type ({type}) when retrieving websocket.
msg.ws.send_error Error while sending data to {type} websockets: {error}.
msg.ws.opened New {type} websocket opened.
msg.ws.closed {type} websocket closed: {reason}.

msg.bw.cache_malformed Tor's 'GETINFO bw-event-cache' provided malformed output: {output}
msg.bw.cache_success Bandwidth graph has information for the last {duration}

msg.log.read_from_log_file Read {count} entries from tor's log file: {path} (read limit: {read_limit}, runtime: {runtime})
msg.log.unable_read Unable to read log located at {path}: {error}
msg.log.bad_format Log located at {path} has a line that doesn't match the format we expect: {line}
msg.log.unknown_runlevel Log located at {path} has an unrecognized runlevel: {runlevel}
msg.log.bad_timestamp Log located at {path} has a timestamp we don't recognize: {value}

msg.setup.unknown_event_types erebus doesn't recognize the following event types: {event_types} (log 'UNKNOWN' events to see them)
msg.setup.already_looping The loop task to connect to tor is already running: {detail}.

msg.usage.invalid_args One or more invalid arguments: {args}
msg.usage.please_use_help {error} (for usage provide --help)
msg.usage.not_a_valid_address '{address}' isn't a valid IPv4 address
msg.usage.not_a_valid_port '{port}' isn't a valid port number
msg.usage.invalid_log_flags Unrecognized event flags: {flags}

msg.debug.header
|Erebus {erebus_version} Debug Dump
|Stem Version: {stem_version}
|Python Version: {python_version}
|Platform: {system} ({platform})
|--------------------------------------------------------------------------------
|Erebus Configuration ({erebusrc_path}):
|{erebusrc_content}
|--------------------------------------------------------------------------------

msg.misc.event_types
|        d DEBUG      a ADDRMAP           k DESCCHANGED   s STREAM
|        i INFO       f AUTHDIR_NEWDESCS  g GUARD         r STREAM_BW
|        n NOTICE     h BUILDTIMEOUT_SET  l NEWCONSENSUS  t STATUS_CLIENT
|        w WARN       b BW                m NEWDESC       u STATUS_GENERAL
|        e ERR        c CIRC              p NS            v STATUS_SERVER
|                     j CLIENTS_SEEN      q ORCONN
|          DINWE tor runlevel+            A All Events
|          12345 erebus runlevel+         X No Events
|                                         U Unknown Events

msg.usage.help_output
|TODO. Please see README for information about usage.

msg.usage.version
|erebus version {version} (released {date})
|

